src/domain/entities/User.ts

```diff
@@ -0,0 +1,16 @@
+export class UserEntity {
+  // Deveria ser User, mas o discord ja tem User, é MUITO chato ficar importando a coisa errada toda hora.
+  constructor(
+    public readonly id: number,
+    public readonly discordId: string,
+    public readonly username: string,
+    public readonly bot: boolean,
+    public readonly status: number,
```

Comment:
Esse status não deveria ser do tipo enum UserStatus? 
```suggestion
    public readonly status: number,
```